src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh

```diff
@@ -14,17 +14,19 @@ fi
 # as disable it by unsetting the variable.
 VSCODE_SHELL_INTEGRATION=1
 
+# Set HISTFILE to the default location before the user's .zshrc is sourced
+# to prevent the value of HISTFILE from defaulting to VSCODE_ZDOTDIR/.zsh_history
+if [[  "$VSCODE_INJECTION" == "1" ]]; then
+	HISTFILE=$USER_ZDOTDIR/.zsh_history
+fi
+
 # Only fix up ZDOTDIR if shell integration was injected (not manually installed) and has not been called yet
 if [[ "$VSCODE_INJECTION" == "1" ]]; then
 	if [[ $options[norcs] = off  && -f $USER_ZDOTDIR/.zshrc ]]; then
 		VSCODE_ZDOTDIR=$ZDOTDIR
 		ZDOTDIR=$USER_ZDOTDIR
+		# A user's custom HISTFILE location might be set when their .zshrc file is sourced below
 		. $USER_ZDOTDIR/.zshrc
-		ZDOTDIR=$VSCODE_ZDOTDIR
```

Comment:
I think we still need this don't we?

Reply:
We don't have a `.zlogout` currently

![Screenshot 2022-12-08 at 4 38 47 PM](https://user-images.githubusercontent.com/29464607/206582070-0cad6a37-457e-42a8-8d8e-258d27b09ad2.png)

Reply:
If you're sure, but I would recommend pulling this out of the candidate fix just in case it causes a regression

Reply:
ok pretty sure it can only do harm to set ZDOTDIR back to our tmp one in this case, but makes sense and will do